tests/unit_tests/encrypted_file.cpp

```diff
@@ -0,0 +1,54 @@
+#include <boost/filesystem.hpp>
+#include <gtest/gtest.h>
+
+#include "crypto/chacha.h"
+#include "seraphis_wallet/encrypted_file.h"
+#include "serialization/serialization.h"
+#include "serialization/keyvalue_serialization.h"
+
+struct test_s
+{
+    std::string data;
+
+    BEGIN_SERIALIZE()
+    FIELD(data)
+    END_SERIALIZE()
+
+    BEGIN_KV_SERIALIZE_MAP()
+    KV_SERIALIZE_VAL_POD_AS_BLOB_FORCE(data)
+    END_KV_SERIALIZE_MAP()
+};
+
+TEST(EncryptedFile, ReadWriteBlob)
+{
+    boost::filesystem::path temp_file = boost::filesystem::unique_path();
```

Comment:
Other test cases seem to use the following construct because the method you use here only returns a file name, and then maybe it's not clear *where* the file gets created, but I must confess that I don't know much about that stuff:

     boost::filesystem::temp_directory_path() / boost::filesystem::unique_path()